### PR TITLE
ENH: Allow "cuda" if requested

### PR DESCRIPTION
**Pull request by larsoner**

### CODE REVIEW

**Comment by dengemann** [comment]
... it seems you've got to rebase...

**Comment by larsoner** [comment]
Rebased.

**Comment by dengemann** [comment]
sorry, you've got to manually merge my fix on @agramfort's contrib.

File: mne/parallel.py [hunk 1]
```diff
@@ -55,31 +55,41 @@ def parallel_func(func, n_jobs, verbose=None):
...
+        Allow n_jobs to be 'cuda'. Default: False.
 
     Returns
     -------
     n_jobs : int
-        The checked number of jobs. Always positive.
+        The checked number of jobs. Always positive (or 'cuda' if
+        applicable.)
     """
-    if n_jobs < 0:
```
Comment (dengemann) [inline mne/parallel.py:71]: i think we need this check.
if n_jobs > 0 n_jobs + n_cores is wrong.

File: mne/parallel.py [hunk 2]
```diff
@@ -55,31 +55,41 @@ def parallel_func(func, n_jobs, verbose=None):
...
+        #else, we have n_jobs='cuda' and this is okay, so do nothing
+    else:
         try:
             import multiprocessing
             n_cores = multiprocessing.cpu_count()
-            if n_cores + n_jobs <= 0:
-                raise ValueError('If n_jobs has a negative value it must not be less '
-                                 'than the number of CPUs present. You\'ve got '
-                                 '%s CPUs' % n_cores)
             n_jobs = n_cores + n_jobs
```
Comment (dengemann) [inline mne/parallel.py:84]: will be wrong for njobs. also this changes the test output (tries to run 3 processes on my 2 cpu machine)

**Comment by larsoner** [comment]
Might it be easier to merge and fix?

**Comment by dengemann** [comment]
Actually it's really easy, you just have to wrap the entire second block in a conditional expecting n_jobs to be < 1.
But i can do it.

**Pull request merged**